tachyon/zk/plonk/constraint_system/constraint_system_unittest.cc

```diff
@@ -37,7 +37,155 @@ TEST_F(ConstraintSystemTest, EnableConstant) {
+          "lookup",
+          [&simple_selector, advice0_copy = advice[0],
+           table0_copy = table[0]](VirtualCells<math::GF7>& cells) {
+            std::unique_ptr<Expression<math::GF7>> simple_selector_expr =
+                cells.QuerySelector(simple_selector);
+            std::unique_ptr<Expression<math::GF7>> advice0_expr =
+                cells.QueryAdvice(advice0_copy, Rotation::Cur());
+
+            LookupPairs<std::unique_ptr<Expression<math::GF7>>,
+                        LookupTableColumn>
+                lookup_pairs;
+            lookup_pairs.emplace_back(
+                std::move(simple_selector_expr) * std::move(advice0_expr),
+                std::move(table0_copy));
+            return lookup_pairs;
+          }),
+      "expression containing simple selector supplied to lookup argument");
+
+  Selector complex_selector = constraint_system.CreateComplexSelector();
+  EXPECT_EQ(
+      constraint_system.Lookup(
+          "lookup",
+          [&complex_selector, &advice, &table](VirtualCells<math::GF7>& cells) {
+            std::unique_ptr<Expression<math::GF7>> complex_selector_expr =
+                cells.QuerySelector(complex_selector);
+            std::unique_ptr<Expression<math::GF7>> advice0_expr =
+                cells.QueryAdvice(advice[0], Rotation::Cur());
+
+            LookupPairs<std::unique_ptr<Expression<math::GF7>>,
+                        LookupTableColumn>
+                lookup_pairs;
+            lookup_pairs.emplace_back(
+                std::move(complex_selector_expr) * std::move(advice0_expr),
+                std::move(table[0]));
+            return lookup_pairs;
+          }),
+      0);
+
+  EXPECT_EQ(
+      constraint_system.Lookup(
+          "lookup",
+          [&complex_selector, &advice, &table](VirtualCells<math::GF7>& cells) {
+            std::unique_ptr<Expression<math::GF7>> complex_selector_expr =
+                cells.QuerySelector(complex_selector);
+            std::unique_ptr<Expression<math::GF7>> advice1_expr =
+                cells.QueryAdvice(advice[1], Rotation::Cur());
+            std::unique_ptr<Expression<math::GF7>> not_complex_selector_expr =
+                ExpressionFactory<math::GF7>::Constant(math::GF7::One()) -
+                complex_selector_expr->Clone();
+            std::unique_ptr<Expression<math::GF7>> default_expr =
+                ExpressionFactory<math::GF7>::Constant(math::GF7(2));
+
+            LookupPairs<std::unique_ptr<Expression<math::GF7>>,
+                        LookupTableColumn>
+                lookup_pairs;
+            lookup_pairs.emplace_back(
+                std::move(complex_selector_expr) * std::move(advice1_expr) +
+                    std::move(not_complex_selector_expr) *
+                        std::move(default_expr),
+                std::move(table[1]));
+            return lookup_pairs;
+          }),
+      1);
+
+  EXPECT_EQ(constraint_system.ComputeLookupRequiredDegree(), 5);
```

Comment:
Could you add checking `lookups()` contain expected elements?

Reply:
@chokobole Oh I should've checked that as well. Thanks for mentioning. Will apply.